src/utils/dataset.py

```diff
@@ -0,0 +1,25 @@
+from dataclasses import dataclass, fields
+from typing import Any
+
+
+def index_by(data_list: list[dataclass], field_str: str) -> dict[Any, dataclass]:
```

Comment:
So it is possible that we should eventually import the pandas python package into this project. It is quite a heavy weight dependency so I have hesitated to bring it in so far. Essentially is is a data analytics package that allows you to use structures called data frames which can be accessed and manipulated easily the way we would want with all our "record sets"  in particular indexing record sets by a specified column as we are doing here. 

While this is a library I am vaguely familiar with, I think it would be a larger refactor that we should probably evaluate the benefits of and consider doing earlier rather than later in this project.

That being said, if this project truest only serves the one purpose we have laid out (solver reimbursements), I might argue that we have nearly already accomplished this without relying on such heavy machinery. 

What do you think?

Reply:
Yeah, let's keep it as it is. In the next repo, we can work with pandas. I don't think that they "heavyness" is a big disadvantage

Reply:
I guess it really depends on how CI is configured. Installing pandas on every commit seems wasteful for little benefit.

Reply:
ONe can just cache it, yes